packages/webpack/src/loader.ts

```diff
@@ -0,0 +1,86 @@
+import * as path from 'path';
+import * as ts from 'typescript';
+import {NgcWebpackPlugin} from './plugin';
+
+// TODO: move all this to ast-tools.
+function _findDecorators(sourceFile: ts.SourceFile, node: ts.Node): ts.Decorator[] {
+  if (node.kind == ts.SyntaxKind.Decorator) {
+    return [node as ts.Decorator];
+  }
+  return node.getChildren(sourceFile).reduce((result, n) => {
+    return result.concat(_findDecorators(sourceFile, n));
+  }, []) as ts.Decorator[];
+}
+
+function _removeDecorators(fileName: string, source: string): string {
+  const sourceFile = ts.createSourceFile(fileName, source, ts.ScriptTarget.Latest);
+  // Find all decorators.
+  const decorators = _findDecorators(sourceFile, sourceFile);
+  decorators.sort((a, b) => b.pos - a.pos);
+
+  decorators.forEach(d => {
+    source = source.slice(0, d.pos) + source.slice(d.end);
+  });
+
+  return source;
+}
+
+
+function _replaceBootstrap(fileName: string, source: string, plugin: NgcWebpackPlugin) {
+  const basePath = path.normalize(plugin.angularCompilerOptions.basePath);
+  const genDir = path.normalize(plugin.genDir);
+  const dirName = path.normalize(path.dirname(fileName));
+  const entryModuleFileName = path.normalize(plugin.entryModule.split('#')[0] + '.ngfactory');
+  const relativeEntryModulePath = path.relative(basePath, entryModuleFileName);
+  const fullEntryModulePath = path.resolve(genDir, relativeEntryModulePath);
+  const relativeNgFactoryPath = path.relative(dirName, fullEntryModulePath);
+  const ngFactoryPath = './' + relativeNgFactoryPath.replace(/\\/g, '/');
+
+  return source
+    .replace(/(import.*)\bplatformBrowserDynamic\b(.*\/)platform-browser-dynamic(\b.*)$/m,
+             '$1 platformBrowser $2platform-browser$3')
+    .replace(/^([\s\S]*)platformBrowserDynamic([\s\S]*).bootstrapModule([\s\S]*)Module([\s\S]*)$/m,
+             '$1platformBrowser$2.bootstrapModuleFactory$3ModuleNgFactory$4')
+    .replace(/^(import.*\w*)Module\b(.*) from .*$/m,
+             `$1ModuleNgFactory$2 from '${ngFactoryPath}';`);
+}
+
+
+// Super simple TS transpiler loader for testing / isolated usage. does not type check!
+export function ngcLoader(source: string) {
+  const plugin = this._compilation._ngToolsWebpackPluginInstance as NgcWebpackPlugin;
```

Comment:
This may be fine for now and once we drop webpack v2. I'm not sure when we are going to start giving deprecation warnings though for trying to access `_compilation`. 

cc @sokra when he gets back from vacation. Maybe we need to have a way to supress warnings of these deprecations of `this._compilation` being unavailable from loaders on a per-plugin basis. That way this usage is acceptable for now, but in the future we can tail it off.